app/helpers/sessions_helper.rb

```diff
@@ -36,4 +36,17 @@ def forget user
     cookies.delete(:user_id)
     cookies.delete(:remember_token)
   end
+
+  def current_user? user
+    user == current_user
+  end
+
+  def redirect_back_or default
+    redirect_to(session[:forwarding_url] || default)
+    session.delete(:forwarding_url)
+  end
+
+  def store_location
+    session[:forwarding_url] = request.original_url if request.get?
+  end
```

Comment:
giải thích chị nghe tại sao cần định nghĩa 2 method này nhé

Reply:
- 2 method dùng cho trường hợp khi user chưa log in mà truy cập vào trang A cần log in, thì sau khi log in chúng ta sẽ forward user tới trang A.
- store_ location sẽ lưu url vào session thông request object với điều kiện method của request phải là GET, nếu là các action POST, PUT, DELETE sẽ gây ra lỗi vì lúc redirect sẽ đưa ra GET request
- redirect_back_or sẽ redirect user tới trang A, nếu không thì sẽ đưa đến đường dẫn default. Sau đó phải xóa url được lưu trong session để login sau đó không bị chuyển tới trang A nếu không nhu cầu.

Reply:
:v chị mún nói đến là do tính chất phi trạng thái của restful, nên phải set url request vào session chú nhé